src/snek5000_cbox/cbox.usr.f

```diff
@@ -67,111 +46,178 @@ subroutine userq  (ix,iy,iz,ieg)
+      call hpts()
+      else
+
+      call opcopy(vtmp(1,1),vtmp(1,2),vtmp(1,ndim),vx,vy,vz)
+      n = NX1*NY1*NZ1*NELV
+      call copy(ttmp,T,n)
+      call opcopy(vx, vy, vz, vxp, vyp, vzp)
+      call copy(T,TP,n)
+
+      call hpts()
+
+      call opcopy(vx,vy,vz, vtmp(1,1),vtmp(1,2),vtmp(1,ndim))
+      call copy(T,ttmp,n)
+
+      endif
       endif
 
       return
       end
 !-----------------------------------------------------------------------
       subroutine userbc (ix,iy,iz,iside,ieg)
       include 'SIZE'
-      include 'TSTEP'
+      include 'GEOM'            
       include 'INPUT'
+      include 'SOLN'            
       include 'NEKUSE'
-      common /rayleigh_r/ rapr,ta2pr
+      
+      integer ntot
+      real delta_T_lateral, delta_T_vertical
+      real xmax, ymax
+
+      delta_T_lateral = UPARAM(5)
+      delta_T_vertical = UPARAM(6)
+      
+      ntot = nx1*ny1*nz1*nelt
+
+      xmax = glmax(xm1,ntot)
+      ymax = glmax(ym1,ntot)
 
-      ux=0.
-      uy=0.
-      uz=0.
+      !base flow
+      if (JP.eq.0) then
+      ux = 0.
+      uy = 0.
+      uz = 0.
 
+      if (delta_T_lateral.ne.0 .and. delta_T_vertical.eq.0) then      
       if (x.eq.0) then
-         temp=-0.5000
-      elseif (x .eq. 1.0) then
-         temp=0.5000
+      temp = -delta_T_lateral/2.
+      elseif (x .eq. xmax) then
+      temp = delta_T_lateral/2.
+      endif
+
+      elseif (delta_T_vertical.ne.0 .and. delta_T_lateral.eq.0) then      
+      if (y.eq.0) then
+      temp = delta_T_vertical/2.
+      elseif (y .eq. ymax) then
+      temp = -delta_T_vertical/2.
+      endif
+      
+      elseif (delta_T_lateral.ne.0 .and. delta_T_vertical.ne.0) then
+      if (x.eq.0) then
+      temp = -delta_T_lateral/2.
+      elseif (x .eq. xmax) then
+      temp = delta_T_lateral/2.      
+      elseif(y.eq.0) then
+      temp = delta_T_vertical/2.
+      elseif (y .eq. ymax) then
+      temp = -delta_T_vertical/2.
+      endif
+      endif
+
+      !perturbation
+      else
+      ux = 0.
+      uy = 0.
+      uz = 0.
+      temp = 0.
       endif
 
       return
       end
 !-----------------------------------------------------------------------
       subroutine useric (ix,iy,iz,ieg)
       include 'SIZE'
-      include 'TOTAL'
       include 'NEKUSE'
+      include 'SOLN'            
+      include 'GEOM'            
+      include 'INPUT'
+
+      integer ntot
+      real delta_T_lateral, delta_T_vertical, amplitude
+      real xmax, ymax
+
+      delta_T_lateral = UPARAM(5)
+      delta_T_vertical = UPARAM(6)
+      amplitude = UPARAM(7)
+
+      ntot = nx1*ny1*nz1*nelt
+
+      xmax = glmax(xm1,ntot)
+      ymax = glmax(ym1,ntot)
+
+      !base flow
+      if (JP.eq.0) then
+      ux = 0.
+      uy = 0.
+      uz = 0.
+      !temp = 0.
+      call random_number(temp)
+      temp = amplitude * temp
+      if (delta_T_vertical.ne.0 .and. delta_T_lateral.eq.0) then
+      temp = temp + delta_T_vertical * (y/ymax - 0.5)
```

Comment:
This file should be correctly indented. I think indenting with 3 spaces is common in Fortran 77.